crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -1442,6 +1442,9 @@ def non_disruptive_upgrade_compute_nodes(names)
           ::Node.find_node_by_name_or_alias(name)
         end
 
+        # remove nodes which were already upgraded (it might leave an empty list)
+        compute_nodes.reject!(&:upgraded?)
```

Comment:
Could you add a log message (and exit from the method) after you check that the list is empty? It does not have to be _completely_ silent

Reply:
done